### PR TITLE
[DEMO] examples: Dont use alpine-dev image for examples

### PR DESCRIPTION
**Pull request by phlax**
Commit Message: DEMO: please ignore
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .azure-pipelines/pipelines.yml [hunk 1]
```diff
@@ -191,7 +191,7 @@ stages:
               AZP_BRANCH: $(Build.SourceBranch)
 
   - stage: linux_x64
-    dependsOn: ["precheck"]
+    dependsOn: []  #  "precheck"]
```
Comment (phlax) [inline .azure-pipelines/pipelines.yml:194]: this is just to speed up running `verify_examples`

File: examples/cache/Dockerfile-service [hunk 2]
```diff
@@ -7,4 +7,7 @@ COPY ./start_service.sh /usr/local/bin/start_service.sh
 COPY ./service-envoy.yaml /etc/service-envoy.yaml
 COPY ./service.py /code
 RUN chmod u+x /usr/local/bin/start_service.sh
-ENTRYPOINT /usr/local/bin/start_service.sh
+
+CMD ["python3", "/code/service.py"]
+
+# ENTRYPOINT /usr/local/bin/start_service.sh
```
Comment (phlax) [inline examples/cache/Dockerfile-service:13]: probs we can remove `start_service.sh` too

File: examples/cache/Dockerfile-service [hunk 3]
```diff
@@ -7,4 +7,7 @@ COPY ./start_service.sh /usr/local/bin/start_service.sh
 COPY ./service-envoy.yaml /etc/service-envoy.yaml
```
Comment (phlax) [inline examples/cache/Dockerfile-service:7]: no need for this either

Comment (oleksiyp) [reply examples/cache/Dockerfile-service:7]: so generally you want to split Flask based service and envoy, right?

Comment (phlax) [reply examples/cache/Dockerfile-service:7]: yep

File: examples/cache/Dockerfile-service (hunk 2 above)
Comment (daixiang0) [reply examples/cache/Dockerfile-service:13]: It includes Envoy cmd ...

File: examples/cache/start_service.sh [hunk 4]
```diff
@@ -1,3 +1,5 @@
 #!/bin/sh
 python3 /code/service.py &
-envoy -c /etc/service-envoy.yaml --service-cluster "service${SERVICE_NAME}"
```
Comment (daixiang0) [inline examples/cache/start_service.sh:3]: It needs to run a envoy process, I am trying to remove it.

**Comment by phlax** [comment]
closing this - ive implemented required changes myself in #16717

**Pull request closed without merging**